tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/ProjectPropertiesRuleTests.cs

```diff
@@ -12,27 +12,29 @@ public sealed class ProjectPropertiesLocalizationRuleTests : XamlRuleTestBase
         public void CategoriesShouldBeDefinedOnFile(string ruleName, string fullPath)
         {
             XElement rule = LoadXamlRule(fullPath, out var namespaceManager);
-            IEnumerable<string> categories = GetCategories();
+            var categories = new HashSet<string>();
+            AddCategories();
+            
             var propertyElements = rule.XPathSelectElements(@"/msb:Rule", namespaceManager).Elements();
 
             // If the page is an extension, we have to check the base page for categories.            
             if (ruleName.Contains(".CSharp"))
-                categories.Union(GetCategories(".CSharp")).Distinct();
-            
+                AddCategories(".CSharp");
+
             else if (ruleName.Contains(".VisualBasic"))
-                categories.Union(GetCategories(".VisualBasic")).Distinct();
-            
+                AddCategories(".VisualBasic");
+
             else if (ruleName.Contains(".FSharp"))
-                categories.Union(GetCategories(".FSharp")).Distinct();
+                AddCategories(".FSharp");
 
             foreach (XElement element in propertyElements)
             {
                 // Skip these xml elements.
-                if (element.Name.LocalName == "Rule.Categories" || element.Name.LocalName == "Rule.DataSource" || element.Name.LocalName == "Rule.Metadata")
+                if (element.Name.LocalName is "Rule.Categories" or "Rule.DataSource" or "Rule.Metadata")
                     continue;
 
                 // Skip if the property is not visible.
-                if (element.Attribute("Visible") is null || element.Attribute("Visible").Value == "False")
+                if (element.Attribute("Visible") is null || bool.TryParse(element.Attribute("Visible").Value, out bool isVisible) )
```

Comment:
This looks wrong. This skips when the value can be parsed.

Is this what you mean?

```c#
if (!bool.TryParse(element.Attribute("Visible")?.Value, out bool isVisible) || !isVisible)
```

That is, continue if it's not parseable, or if it is parseable but not visible.